runtime/sam/expr/eval.go

```diff
@@ -668,8 +668,15 @@ func (i *Index) Eval(ectx Context, this super.Value) super.Value {
 }
 
 func indexArrayOrSet(sctx *super.Context, ectx Context, inner super.Type, vector zcode.Bytes, index super.Value) super.Value {
+again:
+	if index.IsError() {
+		return index
+	}
 	id := index.Type().ID()
-	if !super.IsInteger(id) {
+	if super.IsUnsigned(id) {
+		index = LookupPrimitiveCaster(sctx, super.TypeInt64).Eval(ectx, index)
+		goto again
+	} else if !super.IsInteger(id) {
 		return sctx.WrapError("index is not an integer", index)
 	}
```

Comment:
Nit: Goto is gross. Maybe do something like this?

```go
	id := index.Type().ID()
	if super.IsUnsigned(id) {
		index = LookupPrimitiveCaster(sctx, super.TypeInt64).Eval(ectx, index)
		id = index.Type().ID()
	}
	if !super.IsInteger(id) {
		if index.IsError() {
			return index
		}
		return sctx.WrapError("index is not an integer", index)
	}
```

Reply:
Nice